content/en/docs/concepts/security/mutating-webhooks-good-practices.md

```diff
@@ -0,0 +1,585 @@
+---
+title: Mutating Webhook Good Practices
+description: >
+  Recommendations for writing mutating admission webhooks in Kubernetes.
+content_type: concept
+weight: 60
+---
+
+<!-- overview -->
+
+This page provides good practices and considerations when designing
+_mutating admission webhooks_ in Kubernetes. This information is intended for
+cluster operators who run your own admission webhook servers or third-party
+applications that modify your API requests.
+
+Before reading this page, ensure that you're familiar with the following
+concepts:
+
+* [Admission controllers](/docs/reference/access-authn-authz/admission-controllers/)
+* [Admission webhooks](/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks)
+
+<!-- body -->
+
+## Importance of good webhook design {#why-good-webhook-design-matters}
+
+Mutating admission control occurs when any create, update, or delete request
+is sent to the Kubernetes API. These webhooks are often written to ensure that
+specific fields in object specifications exist or have specific allowed values.
+
+With every release, Kubernetes adds or modifies the API with new features,
+feature promotions to beta or stable status, and deprecations. Even stable
+Kubernetes APIs are likely might change. For example, the `Pod` API changed in
+v1.29 to add the
+[Sidecar containers](/docs/concepts/workloads/pods/sidecar-containers/) feature.
+
+Webhooks that worked as expected with earlier versions of an API might not be
+able to reconcile more recent changes to that API. This can result in unexpected
+behavior after you upgrade your clusters to newer versions.
+
+## Identify whether you use mutating webhooks {#identify-mutating-webhooks}
+
+Even if you don't run your own mutating admission webhooks, some third-party
+applications that you run in your clusters might include mutating webhooks.
+
+To check whether your cluster has any mutating webhooks, run the following
+command:
+
+```shell
+kubectl get mutatingwebhookconfigurations
+```
+The output lists any mutating admission controllers in the cluster. 
```

Comment:
We used to support configuring these statically by passing a plugin configuration file to the kube-apiserver.

At technical review time **or earlier**, let's validate the the old method is completely omitted. Otherwise, this text isn't 100% accurate.

Reply:
Is this the bit that's listed here: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#authenticate-apiservers? Or is that different?